File: server/express/src/modules/user/user.service.ts
import { logger } from "../../config/logger";
import { ApiError } from "../../internal/api.error";
import { userRepository, UserRepositoryInterface } from "./user.repository";

export class UserService {

  constructor(private readonly userRepository: UserRepositoryInterface) {}

  async findUserById(id: number) {
    logger.debug(`Finding user with id ${id}`);
    logger.debug("Unknown") 
    const user = await this.userRepository.findById(id);
    if (!user) {
      throw new ApiError(404,"User not found");
    }
    return user;
  }
  
}

const userService = new UserService(userRepository);
export default userService;

File: server/gofiber/internal/modules/user/user_service.go
package user

import "gofiber/pkg/logger"

type UserServiceInterface interface {
	ProfileUser(id uint) (*User, error)
}

type UserService struct {
	userRepository UserRepositoryInterface
}

func NewUserService(userRepository UserRepositoryInterface) *UserService {
	return &UserService{
		userRepository: userRepository,
	}
}

func (s *UserService) ProfileUser(id uint) (*User, error) {
	logger.Log.Debug("ProfileUser Service Excute")
	user, err := s.userRepository.FindByID(id)
	if err != nil {
		return nil, err
	}
	return user, nil
}
